index/secondary_index_manager.cc

```diff
@@ -116,15 +134,33 @@ view_ptr secondary_index_manager::create_view_for_index(const index_metadata& im
         }
         builder.with_column(index_target->name(), index_target->type, column_kind::clustering_key);
     } else {
-        builder.with_column(index_target->name(), index_target->type, column_kind::partition_key);
+        if (target_type == cql3::statements::index_target::target_type::regular_values) {
+            builder.with_column(index_target->name(), index_target->type, column_kind::partition_key);
+        } else {
+            bytes key_column_name = get_available_computed_collection_column_name(*schema);
+            column_computation_ptr collection_column_computation_ptr = [&name = index_target->name(), target_type] {
+                switch (target_type) {
+                    case cql3::statements::index_target::target_type::keys:
+                        return collection_column_computation::for_keys(name);
+                    case cql3::statements::index_target::target_type::collection_values:
+                        return collection_column_computation::for_values(name);
+                    case cql3::statements::index_target::target_type::keys_and_values:
+                        return collection_column_computation::for_entries(name);
+                    default: abort();
```

Comment:
let's convert this abort() to an assert() with some descriptive message, or, better yet, throwing a logic_error

Reply:
switch cases are caught by the compiler, so it's okay to abort (can only get here if the enum value is out of bounds)